### PR TITLE
mock queryRenderedFeatures

### PR DESCRIPTION
**Pull request by mcwhittemore**
adds simple user based mocking for queryRenderedFeatures.

### CODE REVIEW

**Comment by mcwhittemore** [comment]
@davidtheclark can you give this a review?

File: classes/map.js [hunk 1]
```diff
@@ -92,6 +97,37 @@ Map.prototype.dragPan = {
...
+  features = features.filter(feature => {
+      if (feature.geometry.type === 'Point') {
+        feature = buffer(feature, .00000001, 'kilometers');
+      }
+      var fpoly = envelope({
+        type: 'FeatureCollection',
+        features: [feature]
+      });
+      var merged = turfUnion(fpoly, bpoly);
+      return merged.geometry.type === 'Polygon';
```
Comment (davidtheclark) [inline classes/map.js:166]: I do not understand what is happening with the envelope, union, and filtering out non-Polygons. Could you explain?

**Comment by mcwhittemore** [comment]
@davidtheclark reading the code over, removing envelope should make this a more accurate mock. I have rewritten this and added a handful of comments to try and explain what is going on.

**Comment by davidtheclark** [comment]
Cool, looks good. If it's working the way you intend in your Draw tests, then merge away.

**Pull request merged**